source/egg/math/Quat.cc

```diff
@@ -35,4 +43,39 @@ Vector3f Quatf::rotateVector(const Vector3f &vec) const {
     return res.v;
 }
 
+Quatf Quatf::slerpTo(const Quatf &q1, f32 t) const {
+    f32 dot_ = std::max(-1.0f, std::min(1.0f, dot(q1)));
+    bool bDot = dot_ < 0.0f;
+    dot_ = std::abs(dot_);
+
+    f32 acos = Mathf::acos(dot_);
+    f32 sin = Mathf::sin(acos);
+
+    f32 s;
+    if (std::abs(sin) < 0.00001f) {
+        s = 1.0f - t;
+    } else {
+        f32 invSin = 1.0f / sin;
+        f32 tmp0 = t * acos;
+        s = invSin * Mathf::sin(acos - tmp0);
+        t = invSin * Mathf::sin(tmp0);
+    }
```

Comment:
I'm currently debating whether or not to standardize the `Mathf` namespace within EGG - I'm leaning towards standardizing it, as `max`, `min`, and `abs` are trivially easy to implement into `Mathf`, whereas `sin` and `sqrt` cannot be implemented into `std`. I'm interested in hearing more thoughts.